src/app/personside/dialogpanel/HentOppgaveKnapp.tsx

```diff
@@ -56,117 +51,77 @@ const PLUKKBARE_TEMAGRUPPER = [
-                            Velg temagruppe
-                        </option>
-                        {temagruppeOptions}
-                    </Select>
-                    <KnappBase
-                        id="hentoppgaveknapp"
-                        type="hoved"
-                        onClick={this.onPlukkOppgaver}
-                        spinner={isLoading(this.props.oppgaveResource)}
-                    >
-                        Hent
-                    </KnappBase>
-                </KnappLayout>
-                <ResourceFeilmelding resource={this.props.oppgaveResource} />
-                {tomtTilbakemelding}
-            </HentOppgaveLayout>
+        dispatch(
+            oppgaveResource.actions.post({}, response => {
+                const fødselsnummer = selectFodselsnummerfraOppgaver(response);
+                if (!fødselsnummer) {
+                    setTomKø(true);
+                    return;
+                }
+                setNyBrukerIPath(props.history, fødselsnummer);
+            })
         );
-    }
-
-    private onTemagruppeChange(event: ChangeEvent<HTMLSelectElement>) {
-        this.props.velgTemagruppe(event.target.value);
-        this.setState({ temagruppeFeilmelding: undefined });
-    }
-}
-
-function mapStateToProps(state: AppState, routeProps: RouteComponentProps<{}>): StateProps {
-    return {
-        valgtTemagruppe: state.temagruppe.valgtTemagruppe,
-        oppgaveResource: state.restResources.oppgaver,
-        routeHistory: routeProps.history
     };
-}
 
-function mapDispatchToProps(dispatch: AsyncDispatch): DispatchProps {
-    return {
-        plukkOppgaver: temagruppe => dispatch(plukkOppgaver(temagruppe)),
-        velgTemagruppe: temagruppe => dispatch(velgTemagruppe(temagruppe))
+    const onTemagruppeChange = (event: ChangeEvent<HTMLSelectElement>) => {
+        velgTemaGruppe(event.target.value);
+        setTemaGruppeFeilmelding(false);
     };
+
+    const tomtTilbakemelding = tomKø ? (
+        <AlertStripeInfo>Det er ingen nye oppgaver på valgt temagruppe</AlertStripeInfo>
+    ) : null;
+    const temagruppeOptions = PLUKKBARE_TEMAGRUPPER.map(temagruppe => (
+        <option value={temagruppe.kode} key={temagruppe.kode}>
+            {temagruppe.beskrivelse}
+        </option>
+    ));
+    return (
+        <HentOppgaveLayout>
+            <h2 className="sr-only">Hent oppgave</h2>
+            <KnappLayout>
+                <Select
+                    label="Hent oppgave fra temagruppe"
+                    value={valgtTemaGruppe || ''}
+                    onChange={onTemagruppeChange}
+                    feil={temaGruppeFeilmelding ? { feilmelding: 'Du må velge temagruppe' } : undefined}
+                >
+                    <option disabled={true} value={''}>
+                        Velg temagruppe
+                    </option>
+                    {temagruppeOptions}
+                </Select>
+                <KnappBase
```

Comment:
Mer en generelt spørsmål, for ser det flere steder i kodebasen. Men er det en grunn til at det brukes `<KnappBase type="hoved" />` istedetfor `<HovedKnapp />` ?

Reply:
Hehe, nope, er nok bare fordi folk ikke vet at det finnes en `<HovedKnapp/>` :-P